src/pal/src/loader/module.cpp

```diff
@@ -1339,29 +1342,59 @@ static HMODULE LOADLoadLibrary(LPCSTR shortAsciiName, BOOL fDynamic)
     // See if file can be dlopen()ed; this should work even if it's already loaded
     {
         // See GetProcAddress for an explanation why we leave the PAL.
-        PAL_LeaveHolder holder;
-        dl_handle = dlopen(shortAsciiName, RTLD_LAZY | RTLD_NOLOAD); 
-        if (!dl_handle)
+        PAL_LeaveHolder holder; 
+        
+        // Try to dlopen the module with appropriate variations on the name.
+        // P/Invokes are often declared with shortened versions of the real library
+        // name, such as without the extension or without the "lib" prefix.
+        // We try to open with such variations on the original.
+        int endsWithSuffix = FALSE, startsWithPrefixOrIsAbsolute = FALSE;
+        size_t shortAsciiNameLen = 0;
+        int i;
+        for (i = 1; i <= 4; i++) // four cases: original, with prefix, with suffix, with both prefix and suffix
         {
-            dl_handle = dlopen(shortAsciiName, RTLD_LAZY);
-        }
+            const char* filename = NULL;
+            switch (i)
+            {
+                case 1: // original name
+                    filename = shortAsciiName;
+                    break;
+
+                case 2: // with prefix prepended
```

Comment:
No matter what we do here we're defining policy; it's just a question of how much. Also, not all libraries begin with "lib", even on platforms where many do, and we still need to support those.

If the string "name" is supplied, I think it's very reasonable (required?) to say we try:
    name
    namesuffix
    prefixnamesuffix
Are we agreed on that? I can't tell.

Beyond that, there's the question of whether we also support:
    prefixname
That's the case I was asking about here. I'm tempted to say we don't, but I left it in the PR to have the discussion.

I also don't have an opinion on whether to check whether the prefix or suffix already exists and skip them if they do. I'm fine removing those checks, too, if that's the overall desired outcome. I see the arguments for not having them, and they make sense to me.